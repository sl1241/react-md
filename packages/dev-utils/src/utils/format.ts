import log from "loglevel"
import type { BuiltInParserName } from "prettier"
import prettier from "prettier"

function getParser(
    code: string,
    parser: BuiltInParserName | undefined,
    filePath: string | undefined
): BuiltInParserName | undefined {
    if (typeof parser === "string") {
        return parser
    }

    if (typeof filePath === "string") {
        return undefined
    }

    if (code.startsWith("{") || code.startsWith("[")) {
        return "json"
    }

    if (/^(@import|\.[a-z])/m.test(code) || /^\s+@(mixin|include)/.test(code)) {
        return "scss"
    }

    return "typescript"
}

export function format(
    code: string,
    parser?: BuiltInParserName,
    filepath?: FunctionMaybe<Nullable<string>>
): string {
    const resolvedParser = getParser(code, parser, filepath)
    try {
        return prettier.format(code, {
            filepath,
            parser: resolvedParser,
            proseWrap: resolvedParser === "markdown" ? "always" : undefined,
            singleQuote: resolvedParser === "scss" || resolvedParser === "babel",
        })
    } catch (e) {
        log.error(e)
        process.exit(1)
    }
}
