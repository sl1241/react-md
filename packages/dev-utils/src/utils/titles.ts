import { upperFirst } from "lodash"

/**
 * A simple utility that converts some string into a page/section title. This
 * really just capitalizes each word in the string after it has been split by
 * hyphens or capital letters.
 *
 * @param s - The string to convert to a title.
 * @param joinWith - A string that should be used to join all the parts together
 * to create a title.
 * @param capitals - Boolean if the string should be split on capital letters
 * instead of hyphens.
 * @returns A title-ized string.
 */
export function toTitle(s: string, joinWith = " ", capitals = false): string {
    if (/autocomplete/i.test(s)) {
        return "AutoComplete"
    }

    if (/^api$/i.test(s)) {
        return "API"
    }

    if (/sassdoc/i.test(s)) {
        return "SassDoc"
    }

    if (s === "[id]") {
        // TODO: Fix to use query params instead
        return "Layout"
    }

    return s
        .split(capitals ? /(?=[A-Z])/ : "-")
        .map((part) => {
            if (/^((v\d+)|(to))$/.test(part)) {
                return part
            }

            if (/^api$/i.test(part)) {
                return "API"
            }

            if (/^cdn$/i.test(part)) {
                return "CDN"
            }

            return upperFirst(part)
        })
        .join(joinWith)
}

/**
 * Converts a pathname into a breadcrumb type of title to display in the
 * documentation site.
 *
 * @param pathname - The current pathname to create a title for
 * @param statusCode - An optional status code to check against for custom error
 * cases.
 * @returns a string to display in the documentation site's breadcrumbs.
 */
export function toBreadcrumbPageTitle(
    pathname: string,
    statusCode?: FunctionMaybe<Nullable<number>>
): string {
    let title = ""
    if (statusCode) {
        switch (statusCode) {
            case 404:
                title = "Not Found"
                break
            default:
                title = "Server error"
        }
    } else if (/v\d+-to-v\d+$/.test(pathname)) {
        const [migration] = pathname.split("/").reverse()
        title = `Migration Guides - ${migration.replace(/-/g, " ")}`
    } else {
        const parts = pathname.split("/").filter((p) => !!p && !/packages/.test(p))
        title = parts.map((p) => toTitle(p)).join(" - ")
    }

    return `react-md${title ? ` - ${title}` : ""}`
}

/**
 * This is a utility function that will create a unique id for a title string.
 * The title string should normally be somewhere betwen 5-20 characters.
 *
 * This is generally used to create ids for links within the documentation page
 * for headers to generate quick links and table of contents.
 *
 * @param title - The title to create an id for.
 * @returns an id based on the title
 */
export function toId(title: string): string {
    return title
        .replace(/\/|\\|\[|]|-/g, "")
        .replace(/SCSS/g, "scss ")
        .replace(/CSS/g, "css ")
        .replace(/API/g, "api")
        .replace(/\s+/g, " ")
        .trim()
        .split(/\s|(?=[A-Z])/)
        .join("-")
        .toLowerCase()
}
