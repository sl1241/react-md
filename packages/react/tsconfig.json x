{
  "extends": "../../tsconfig.base.json",
  "compilerOptions": {
    "outDir": "./dist",
    "rootDir": "./src",
    "declarationDir": "./dist/types",
    "composite": true,
    "skipLibCheck": true,
    "skipDefaultLibCheck": true,
    "jsx": "preserve",
    "jsxImportSource": "voby",
    "types": [
      "voby/dist/jsx/jsx"
    ],
  },
  "exclude": [
    "vite.config.ts",
"vite.config.umd.ts",
    "types",
  ],
}
