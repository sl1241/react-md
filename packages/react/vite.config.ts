import { defineConfig } from 'vite'

const config = defineConfig({
    build: {
        minify: false,
        lib: {
            entry: ["./src/index.ts"],
            name: "react",
            formats: ['cjs', 'es', 'umd'],
            fileName: (format: string, entryName: string) => `${entryName}.${format}.js`
        },
        // rollupOptions: {
        //   output: {
        //     preserveModules: true,
        //   }
        // },
    },


    esbuild: {
        jsx: 'automatic',
    },

    css: {
        modules: {
            scopeBehaviour: 'global',
            localsConvention: 'camelCase',
        }
    },
})



export default config
