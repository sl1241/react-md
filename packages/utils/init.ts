import '@testing-library/jest-dom'
import '@testing-library/jest-dom/extend-expect'

// import {
//   DEFAULT_DESKTOP_MIN_WIDTH,
//   DEFAULT_DESKTOP_LARGE_MIN_WIDTH,
// } from '../packages/utils/lib/sizing/constants'

// // window will be undefined for the one test I force to be run in node instead
// // of jsdom
// if (typeof window !== 'undefined' && typeof window.matchMedia !== 'function') {
//   window.matchMedia = (query) => ({
//     media: query,
//     matches:
//       query.includes(`${DEFAULT_DESKTOP_MIN_WIDTH}`) ||
//       query.includes(`${DEFAULT_DESKTOP_LARGE_MIN_WIDTH}`),
//     onchange: () => {},
//     addListener: () => {},
//     removeListener: () => {},
//     addEventListener: () => {},
//     removeEventListener: () => {},
//     dispatchEvent: () => false,
//   });
// }
