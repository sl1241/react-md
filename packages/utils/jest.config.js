/** @type {import('ts-jest').JestConfigWithTsJest} */
// module.exports = {
//     preset: 'ts-jest',
//     testEnvironment: 'jsdom',

//     // globals: {
//     //     'ts-jest': {
//     //         compiler: "typescript"
//     //         // ts-jest configuration goes here
//     //     }
//     // },

//     transform: {
//         "^.+\\.tsx?$": "ts-jest",
//     },
//     testRegex: "(/__tests__/.*|(\\.|/)(test|spec))\\.(jsx?|tsx?)$",
//     moduleFileExtensions: ["ts", "tsx", "js", "jsx", "json", "node"],
// }

module.exports = {
  preset: 'ts-jest',
  testEnvironment: 'jsdom',

  setupFilesAfterEnv: ['<rootDir>/init.ts'],

  testMatch: ['**/?(*.)+(spec|test).ts?(x)', '**/__tests__/**/*.ts?(x)'],
  globals: {
    'ts-jest': {
      tsconfig: {
        jsx: 'react-jsx',
        jsxImportSource: 'voby',
        strict: true,
        moduleResolution: 'node',
        // preserveSymlinks: true,
        noImplicitAny: false,
        strictNullChecks: false,
        lib: ['dom', 'dom.iterable', 'esnext'],
        types: ['voby/dist/jsx/jsx', 'jest'],
      },
    },
  },
  // transform: {
  //     // '\\.svg$': '<rootDir>/testSetup/inlineSvgTransformer.js',
  //     // '\\.(jpg|jpeg|png|gif|eot|otf|webp|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$':
  //     //     '<rootDir>/testSetup/fileTransformer.js',
  //     '^.+\\.tsx?$': ['ts-jest', {
  //         // tsconfig: './tsconfig.test.json',
  //         // useESM: true,
  //         // // other ts-jest config options
  //         jsx: 'react-jsx',
  //         jsxImportSource: 'voby/testing',
  //         strict: true,
  //         moduleResolution: 'node',
  //         // preserveSymlinks: true,
  //         noImplicitAny: false,
  //         strictNullChecks: false,
  //         lib: [
  //             'dom',
  //             'dom.iterable',
  //             'esnext',
  //         ],
  //         types: [
  //             'voby/dist/jsx/jsx',
  //             'jest',
  //         ],
  //     }],
  // },

  moduleNameMapper: {
    '^@/(.*)$': '<rootDir>/src/$1',
  },
  moduleFileExtensions: ['js', 'json', 'jsx', 'node', 'ts', 'tsx'],
};
