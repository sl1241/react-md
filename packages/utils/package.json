{
  "name": "@react-md/utils",
  "version": "5.1.3",
  "description": "General utils for react-md.",
  "main": "./dist/index.cjs.js",
  "module": "./dist/index.es.js",
  "browser": "./dist/index.umd.js",
  "types": "./dist/types/index.d.ts",
  "sass": "./dist/styles.scss",
  "sideEffects": [
    "dist/**/*"
  ],
  "scripts": {
    "release": "git add . && git commit -m \"Bump version\" && git push && npm version patch",
    "declaration": "tsc --declaration --emitDeclarationOnly",
    "build:only": "vite build && vite build --config vite.config.umd.ts",
    "build": "run-s build:only declaration",
    "test": "vitest"
  },
  "author": "Mikkel Laursen <mlaursen03@gmail.com>",
  "repository": {
    "type": "git",
    "url": "https://github.com/mlaursen/react-md.git",
    "directory": "packages/utils"
  },
  "bugs": {
    "url": "https://github.com/mlaursen/react-md/issues"
  },
  "homepage": "https://react-md.dev/packages/utils/demos",
  "keywords": [
    "accessibility",
    "a11y",
    "react-md",
    "utils",
    "throttled events",
    "rtl",
    "right to left websites",
    "safari scroll fix"
  ],
  "license": "MIT",
  "dependencies": {
    "@juggle/resize-observer": "^3.4.0",
    "@react-md/react": "workspace:../react",
    "chk": "workspace:../../../chk",
    "jest-web": "workspace:../../../jest/packages/jest-web",
    "classnames": "^2.3.2",
    "csstype": "^3.1.2",
    "pnpm": "^8.3.1",
    "voby": "workspace:../../../woby"
  },
  "devDependencies": {
    "@jest/globals": "^29.5.0",
    "@testing-library/jest-dom": "^5.16.5",
    "@types/jest": "^29.5.1",
    "@types/testing-library__jest-dom": "^5.14.5",
    "@vitest/browser": "^0.30.1",
    "@vitest/ui": "^0.30.1",
    "esbuild-jest": "^0.5.0",
    "jest": "^29.5.0",
    "jest-preview": "^0.3.1",
    "jsdom": "^21.1.2",
    "postcss": "^8.4.23",
    "prettier": "^2.8.8",
    "react2voby": "workspace:../../../react2voby",
    "sass": "^1.62.1",
    "ts-jest": "^29.1.0",
    "typescript": "^5.0.4",
    "vite": "^4.3.3",
    "vite-plugin-sass-dts": "^1.3.4",
    "vite-plugin-static-copy": "^0.13.1",
    "vitest": "^0.30.1",
    "voby-vite": "^1.1.0",
    "webdriverio": "^8.9.0"
  },
  "peerDependenciesMeta": {
    "react": {
      "optional": false
    }
  },
  "publishConfig": {
    "access": "public"
  }
}
