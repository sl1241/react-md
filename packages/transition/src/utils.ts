import { getViewportSize } from "@react-md/utils"
import type {
    CSSTransitionClassNames,
    CSSTransitionClassNamesObject,
    TransitionActions,
    TransitionTimeout,
    TransitionTimeoutObject,
} from "./types"
import { CSSProperties, $$ } from "voby"

/**
 * @remarks \@since 4.0.0
 * @internal
 */
export interface TransitionTimeoutOptions extends Required<TransitionActions> {
    timeout: FunctionMaybe<TransitionTimeout>
}

/**
 * @remarks \@since 4.0.0
 * @internal
 */
export function getTransitionTimeout({ timeout, appear, enter, exit, }: Readonly<TransitionTimeoutOptions>): Readonly<Required<TransitionTimeoutObject>> {
    const to = $$(timeout)

    if (typeof to === "number") {
        return {
            appear: appear ? to : 0,
            enter: enter ? to : 0,
            exit: exit ? to : 0,
        }
    }

    return {
        appear: (appear && (to.appear ?? to.enter)) || 0,
        enter: (enter && to.enter) || 0,
        exit: (exit && to.exit) || 0,
    }
}

/**
 * @remarks \@since 4.0.0
 * @internal
 */
export interface CollapseSizing {
    maxHeight?: CSSProperties['maxHeight']
    paddingTop?: CSSProperties['paddingTop']
    paddingBottom?: CSSProperties['paddingBottom']
}

/**
 * A small util that will find the max-height, padding-top, and padding-bottom
 * for the provided element. This is really used to be able to transition the
 * max-height value since `max-height: auto` does not transition. The only way
 * to get transition is to change max-height values manually.
 *
 * @remarks \@since 4.0.0
 * @internal
 */
export function getElementSizing(element: HTMLElement | null): CollapseSizing {
    let maxHeight
    let paddingTop
    let paddingBottom
    if (element) {
        maxHeight = element.scrollHeight

        // clone the element (not deep) just to figure out it's padding without the
        // inline styles applied
        const cloned = element.cloneNode(false) as HTMLElement
        cloned.style.maxHeight = ""
        cloned.style.padding = ""
        cloned.style.paddingLeft = element.style.paddingLeft
        cloned.style.paddingRight = element.style.paddingRight
        cloned.style.visibility = "hidden"

        const container = element.parentElement || document.body
        container.appendChild(cloned)
        const style = window.getComputedStyle(cloned)
        if (style.paddingTop) {
            paddingTop = parseFloat(style.paddingTop)
        }

        if (style.paddingBottom) {
            paddingBottom = parseFloat(style.paddingBottom)
        }
        container.removeChild(cloned)
    }

    return { maxHeight, paddingTop, paddingBottom }
}

/**
 * @remarks \@since 4.0.0
 * @internal
 */
export interface TransitionClassNamesOptions extends TransitionTimeoutOptions {
    classNames: FunctionMaybe<CSSTransitionClassNames>
}

/**
 * @remarks \@since 4.0.0
 * @internal
 */
export function getTransitionClassNames({ classNames: cs, ...timeoutOptions }: TransitionClassNamesOptions): Readonly<Required<CSSTransitionClassNamesObject>> {
    const classNames = $$(cs)
    const timeout = getTransitionTimeout(timeoutOptions)

    if (typeof classNames === "string") {
        const { appear, enter, exit } = timeout
        return {
            appear: appear ? `${classNames}--appear` : "",
            appearActive: appear ? `${classNames}--appear-active` : "",
            appearDone: "",
            enter: enter ? `${classNames}--enter` : "",
            enterActive: enter ? `${classNames}--enter-active` : "",
            enterDone: "",
            exit: exit ? `${classNames}--exit` : "",
            exitActive: exit ? `${classNames}--exit-active` : "",
            exitDone: "",
        }
    }

    const {
        enter = "",
        enterActive = "",
        enterDone = "",
        exit = "",
        exitActive = "",
        exitDone = "",
        appear = (timeout.appear && enter) || "",
        appearActive = (timeout.appear && enterActive) || "",
        appearDone = (timeout.appear && enterDone) || "",
    } = classNames

    return {
        appear,
        appearActive,
        appearDone,
        enter,
        enterActive,
        enterDone,
        exit,
        exitActive,
        exitDone,
    }
}

/**
 * @remarks \@since 4.0.0
 * @internal
 */
interface IsWithinViewportOptions {
    fixedElement: HTMLElement
    fixedToElement: HTMLElement
}

/**
 * @remarks \@since 4.0.0
 * @internal
 */
export function isWithinViewport({
    fixedElement,
    fixedToElement,
}: IsWithinViewportOptions): boolean {
    const fixedElementRect = fixedElement.getBoundingClientRect()
    const fixedToElementRect = fixedToElement.getBoundingClientRect()
    const vh = getViewportSize("height")
    const vw = getViewportSize("width")
    const top = Math.min(fixedElementRect.top, fixedToElementRect.top)
    const right = Math.max(fixedElementRect.right, fixedToElementRect.right)
    const bottom = Math.max(fixedElementRect.bottom, fixedToElementRect.bottom)
    const left = Math.min(fixedElementRect.left, fixedToElementRect.left)

    return bottom >= 0 && top <= vh && right >= 0 && left <= vw
}
