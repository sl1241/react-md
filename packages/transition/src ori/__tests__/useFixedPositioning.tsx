import type { ReactElement } from "react";
import { useRef, useState } from "react";
import { act, fireEvent, render } from "@testing-library/react";
import { TOP_INNER_RIGHT_ANCHOR } from "@react-md/utils";

import type { FixedPositioningOptions } from "../useFixedPositioning";
import { useFixedPositioning } from "../useFixedPositioning";
import { useScaleTransition } from "../useScaleTransition";
import type { FixedPositioningTransitionCallbacks } from "../types";

interface TestProps
  extends Omit<
    FixedPositioningOptions<HTMLButtonElement, HTMLDivElement>,
    "fixedTo"
  > {
  defaultVisible?: boolean;
}

function Test({ defaultVisible = false, ...options }: TestProps): ReactElement {
  const buttonRef = useRef<HTMLButtonElement>(null);
  const [visible, setVisible] = useState(defaultVisible);
  const { style, transitionOptions } = useFixedPositioning({
    ...options,
    fixedTo: buttonRef,
  });
  const { elementProps, rendered } = useScaleTransition({
    ...transitionOptions,
    transitionIn: visible,
  });

  return (
    <>
      <button
        ref={buttonRef}
        type="button"
        onClick={() => setVisible((p) => !p)}
      >
        Toggle
      </button>
      {rendered && (
        <div {...elementProps} style={style} data-testid="element">
          Some content.
        </div>
      )}
    </>
  );
}

describe("useFixedPositioning", () => {
  beforeAll(() => {
    jest.useFakeTimers();
  });

  afterEach(() => {
    jest.clearAllTimers();
  });

  it("should default to fixing itself with the BELOW_CENTER_ANCHOR", () => {
    const { container, getByRole, getByTestId } = render(<Test />);
    const getElement = (): HTMLElement => getByTestId("element");
    const button = getByRole("button");

    expect(getElement).toThrow();
    expect(container).toMatchSnapshot();

    fireEvent.click(button);
    expect(getElement).not.toThrow();
    expect(container).toMatchSnapshot();

    act(() => {
      jest.runAllTimers();
    });
    expect(container).toMatchSnapshot();

    fireEvent.click(button);
    expect(getElement).not.toThrow();
    expect(container).toMatchSnapshot();

    act(() => {
      jest.runAllTimers();
    });
    expect(getElement).toThrow();
    expect(container).toMatchSnapshot();
  });

  it("should detect if the fixed element is still visible in the viewport on scroll events", () => {
    const onScroll = jest.fn();
    const { getByRole } = render(<Test onScroll={onScroll} />);
    const toggle = getByRole("button");

    fireEvent.click(toggle);
    act(() => {
      jest.runAllTimers();
    });
    expect(onScroll).not.toBeCalled();

    const scrollEvent = new UIEvent("scroll");
    act(() => {
      window.dispatchEvent(scrollEvent);
      jest.runAllTimers();
    });

    // this logic is really handled in the utils tests.
    expect(onScroll).toBeCalledWith(scrollEvent, {
      visible: expect.any(Boolean),
      fixedElement: expect.any(HTMLElement),
      fixedToElement: expect.any(HTMLElement),
    });
  });

  it("should automatically update the style when the page is resized", () => {
    const onResize = jest.fn();
    const { getByRole } = render(<Test onResize={onResize} />);
    const toggle = getByRole("button");

    fireEvent.click(toggle);
    act(() => {
      jest.runAllTimers();
    });
    expect(onResize).not.toBeCalled();

    act(() => {
      window.dispatchEvent(new UIEvent("resize"));
      jest.runAllTimers();
    });
    expect(onResize).toBeCalled();
  });

  it("should update the style correctly based on the initialX and initialY so that it can be used for context menus", () => {
    const { container, getByRole, rerender } = render(<Test />);
    const toggle = getByRole("button");

    fireEvent.click(toggle);
    act(() => {
      jest.runAllTimers();
    });
    expect(container).toMatchSnapshot();

    rerender(<Test initialX={30} initialY={40} />);
    expect(container).toMatchSnapshot();
  });

  it("should correctly call the transition callbacks", () => {
    const onEnter = jest.fn();
    const onEntering = jest.fn();
    const onEntered = jest.fn();
    const onExited = jest.fn();
    const props: Required<FixedPositioningTransitionCallbacks> = {
      onEnter,
      onEntering,
      onEntered,
      onExited,
    };

    const { getByRole } = render(<Test {...props} />);
    const toggle = getByRole("button");

    expect(onEnter).not.toBeCalled();
    expect(onEntering).not.toBeCalled();
    expect(onEntered).not.toBeCalled();
    expect(onExited).not.toBeCalled();

    fireEvent.click(toggle);
    expect(onEnter).toBeCalledTimes(1);
    expect(onEntering).toBeCalledTimes(1);
    expect(onEntered).not.toBeCalled();
    expect(onExited).not.toBeCalled();

    act(() => {
      jest.runAllTimers();
    });
    expect(onEnter).toBeCalledTimes(1);
    expect(onEntering).toBeCalledTimes(1);
    expect(onEntered).toBeCalledTimes(1);
    expect(onExited).not.toBeCalled();

    fireEvent.click(toggle);
    expect(onEnter).toBeCalledTimes(1);
    expect(onEntering).toBeCalledTimes(1);
    expect(onEntered).toBeCalledTimes(1);
    expect(onExited).not.toBeCalled();

    act(() => {
      jest.runAllTimers();
    });
    expect(onEnter).toBeCalledTimes(1);
    expect(onEntering).toBeCalledTimes(1);
    expect(onEntered).toBeCalledTimes(1);
    expect(onExited).toBeCalledTimes(1);
  });

  it("should allow for configuring the fixed position", () => {
    const { container, getByRole } = render(
      <Test transformOrigin anchor={TOP_INNER_RIGHT_ANCHOR} />
    );
    const toggle = getByRole("button");

    fireEvent.click(toggle);
    act(() => {
      jest.runAllTimers();
    });
    expect(container).toMatchSnapshot();
  });
});
