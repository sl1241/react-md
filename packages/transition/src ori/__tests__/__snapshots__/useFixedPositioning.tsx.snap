// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useFixedPositioning should allow for configuring the fixed position 1`] = `
<div>
  <button
    type="button"
  >
    Toggle
  </button>
  <div
    data-testid="element"
    style="left: 16px; top: 0px; position: fixed; transform-origin: 0 0;"
  >
    Some content.
  </div>
</div>
`;

exports[`useFixedPositioning should default to fixing itself with the BELOW_CENTER_ANCHOR 1`] = `
<div>
  <button
    type="button"
  >
    Toggle
  </button>
</div>
`;

exports[`useFixedPositioning should default to fixing itself with the BELOW_CENTER_ANCHOR 2`] = `
<div>
  <button
    type="button"
  >
    Toggle
  </button>
  <div
    class="rmd-transition--scale-enter rmd-transition--scale-enter-active"
    data-testid="element"
    style="left: 16px; top: 0px; position: fixed;"
  >
    Some content.
  </div>
</div>
`;

exports[`useFixedPositioning should default to fixing itself with the BELOW_CENTER_ANCHOR 3`] = `
<div>
  <button
    type="button"
  >
    Toggle
  </button>
  <div
    data-testid="element"
    style="left: 16px; top: 0px; position: fixed;"
  >
    Some content.
  </div>
</div>
`;

exports[`useFixedPositioning should default to fixing itself with the BELOW_CENTER_ANCHOR 4`] = `
<div>
  <button
    type="button"
  >
    Toggle
  </button>
  <div
    class="rmd-transition--scale-exit rmd-transition--scale-exit-active"
    data-testid="element"
    style="left: 16px; top: 0px; position: fixed;"
  >
    Some content.
  </div>
</div>
`;

exports[`useFixedPositioning should default to fixing itself with the BELOW_CENTER_ANCHOR 5`] = `
<div>
  <button
    type="button"
  >
    Toggle
  </button>
</div>
`;

exports[`useFixedPositioning should update the style correctly based on the initialX and initialY so that it can be used for context menus 1`] = `
<div>
  <button
    type="button"
  >
    Toggle
  </button>
  <div
    data-testid="element"
    style="left: 16px; top: 0px; position: fixed;"
  >
    Some content.
  </div>
</div>
`;

exports[`useFixedPositioning should update the style correctly based on the initialX and initialY so that it can be used for context menus 2`] = `
<div>
  <button
    type="button"
  >
    Toggle
  </button>
  <div
    data-testid="element"
    style="left: 30px; top: 40px; position: fixed;"
  >
    Some content.
  </div>
</div>
`;
